### DISCUSSION TITLE
whisper.transcribe() Fails on Multi-GPU Systems: Incorrect Tensor Transfer to CUDA Device

### DISCUSSION THREAD

**Discussion by jaimebs2** [openai/whisper#2382] [General]
I’ve encountered a bug in the _dtw_cuda()_ function located in timing.py. The issue arises when the cost tensor is sent to _.cuda()_ without considering the device on which the input tensor x resides.

In the function _dtw_cuda()_, the cost tensor is being sent to _.cuda()_ directly, which defaults to _"cuda:0"_. This causes an issue if the tensor x is on a different device, such as _"cuda:1"_, resulting in a device mismatch error. The cost tensor should be sent to the device where x is located (i.e., x.device) instead of assuming it is on _"cuda:0"_.

If x is on _"cuda:1"_ it will raise the following error:

`ValueError: Pointer argument (at 2) cannot be accessed from Triton (cpu tensor?)`

**Solution:**
Change _dtw_cuda()_ function.

`cost = cost.to(device=x.device)`

**Comment by lukebelbina** [comment]
I am also running into this issue
